### PR TITLE
Add env and collection to DB context, fix names of env vars in local-proxy

### PR DESCRIPTION
**Pull request by bergundy**

### CODE REVIEW

File: db-client/src/test/component/db.test.ts [hunk 1]
```diff
@@ -43,6 +43,8 @@ test.beforeEach(async (t) => {
     `http://localhost:${port}/v1`,
     {
       appId: 'testing',
+      appEnv: 'default',
+      collection: 'default',
```
Comment (vogre) [inline db-client/src/test/component/db.test.ts]: Might be confusing that both env and collection are default...

File: db-client/src/index.ts [hunk 2]
```diff
@@ -9,14 +9,23 @@ import { DB, Versioned, Q } from './db';
 
 export { Q };
 
-// TODO(bergundy): Verify environment variables
+function assertEnv(name: string): string {
```
Comment (vogre) [inline db-client/src/index.ts]: Better than undefineds, although still missing a test of shift-db in local-proxy (not necessarily for this PR).

Comment (bergundy) [reply db-client/src/index.ts]: #172

File: db-client/src/test/component/db.test.ts (hunk 1 above)
Comment (bergundy) [reply db-client/src/test/component/db.test.ts]: Hmm.. that's what we use ATM in the deployed version.

**Pull request merged**